src/hypothesis/searchstrategy/regex.py

```diff
@@ -0,0 +1,376 @@
+# coding=utf-8
+#
+# This file is part of Hypothesis, which may be found at
+# https://github.com/HypothesisWorks/hypothesis-python
+#
+# Most of this work is copyright (C) 2013-2017 David R. MacIver
+# (david@drmaciver.com), but it contains contributions by others. See
+# CONTRIBUTING.rst for a full list of people who may hold copyright, and
+# consult the git log if you need to determine who owns an individual
+# contribution.
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v. 2.0. If a copy of the MPL was not distributed with this file, You can
+# obtain one at http://mozilla.org/MPL/2.0/.
+#
+# END HEADER
+
+from __future__ import division, print_function, absolute_import
+
+import re
+import sys
+import sre_parse as sre
+
+import hypothesis.strategies as hs
+from hypothesis.errors import InvalidArgument
+from hypothesis.internal.compat import PY3, hrange, hunichr, text_type
+
+HAS_SUBPATTERN_FLAGS = sys.version_info[:2] >= (3, 6)
+
+
+UNICODE_CATEGORIES = set([
+    'Cf', 'Cn', 'Co', 'LC', 'Ll', 'Lm', 'Lo', 'Lt', 'Lu',
+    'Mc', 'Me', 'Mn', 'Nd', 'Nl', 'No', 'Pc', 'Pd', 'Pe',
+    'Pf', 'Pi', 'Po', 'Ps', 'Sc', 'Sk', 'Sm', 'So', 'Zl',
+    'Zp', 'Zs',
+])
+
+
+SPACE_CHARS = set(u' \t\n\r\f\v')
+UNICODE_SPACE_CHARS = SPACE_CHARS | set(u'\x1c\x1d\x1e\x1f\x85')
+UNICODE_DIGIT_CATEGORIES = set(['Nd'])
+UNICODE_SPACE_CATEGORIES = set(['Zs', 'Zl', 'Zp'])
+UNICODE_LETTER_CATEGORIES = set(['LC', 'Ll', 'Lm', 'Lo', 'Lt', 'Lu'])
+UNICODE_WORD_CATEGORIES = UNICODE_LETTER_CATEGORIES | set(['Nd', 'Nl', 'No'])
+
+# On Python >= 3.0 and < 3.4 particular unicode word chars are not
```

Comment:
Hmm.  All tests pass when I delete this, so I'm going to assume the comment was correct.